src/profilers.cpp

```diff
@@ -2,19 +2,369 @@
+
+    const int ERROR_STDERR = 201;
+    const int ERROR_STDOUT_DUP2 = 202;
+    const int ERROR_STDERR_DUP2 = 203;
+    const int ERROR_NO_NUMBER_REGEX = 204;
+    const int ERROR_TOO_MANY_NUMBERS_REGEX = 205;
+    const int ERROR_CONVERSION_TO_FLOAT = 206;
+    const int ERROR_PARSING_CONNECTION_INSTRS = 207;
+    
+    
+    std::vector<std::string> parts = boost::program_options::split_unix(this->metric_command);
+
+    const char *path = parts[0].c_str();
+    char *parts_c_str[parts.size() + 1];
+
+    for (int i = 0; i < parts.size(); i++) {
+      parts_c_str[i] = (char *)parts[i].c_str();
+    }
+    parts_c_str[parts.size()] = NULL;
+
+
+    pid_t forked_metric_profiler = fork(); //separate process for profiler, parent handles errors
+
+    if(forked_metric_profiler == 0){
+      int code_metric_exec = 0;
+
+      int status;
+      pid_t command_status = waitpid(pid, &status, WNOHANG); //check if profiled command has finished executing
+      const int interval_ns = 1000 * 1000000 / this->freq;
+
+      while (command_status > 0){
+
+        struct timespec start, end;
+        clock_gettime(CLOCK_MONOTONIC, &start);
+
+        int pipe_fd[2];
+
+        if (pipe(pipe_fd) == -1) {
+          print("Could not establish communication pipe between metric-exec "
+                "and metric-read in \"" + this->get_name() + "\"! Terminating "
+                "the profiled command wrapper.", true, true);
+          kill(pid, SIGTERM);
+          return;
+        }
+
+        pid_t forked_metric_exec = fork();
+        if (forked_metric_exec == 0){
+          close(pipe_fd[0]);
+
+          fs::current_path(result_processed);
+
+          int stderr_fd = creat(stderr_metric_command.c_str(), O_WRONLY);
+
+          if (stderr_fd == -1) {
+            std::exit(ERROR_STDERR);
+          }
+
+          if (dup2(stderr_fd, STDERR_FILENO) == -1) {
+            std::exit(ERROR_STDERR_DUP2);
+          }
+
+          if (dup2(pipe_fd[1], STDOUT_FILENO) == -1) {
+            std::exit(ERROR_STDOUT_DUP2);
+          }
+
+          close(pipe_fd[1]);
+          execvp(path, parts_c_str);
+
+          std::exit(errno);
+        }
+        else
+        {
+          close(pipe_fd[1]);
+          //read
+          char buffer[1024];
+          ssize_t bytes_read;
+          std::string data;
+          
+          while ((bytes_read = read(pipe_fd[0], buffer, sizeof(buffer) - 1)) > 0) {
+              buffer[bytes_read] = '\0';
+              data += buffer;
+          }
+          std::string parsed_data;
+
+          if(!this->regex.empty()){
+
+            std::regex number_regex(this->regex);
+
+            std::sregex_iterator numbers_begin = std::sregex_iterator(data.begin(), data.end(), number_regex);
+            std::sregex_iterator numbers_end = std::sregex_iterator();
+            for (std::sregex_iterator i = numbers_begin; i != numbers_end; ++i) {
+              std::smatch match = *i;
+
+              parsed_data += std::stoi(match.str());
```

Comment:
Also, you should assume that a correct regex has a single capturing group, e.g. "blabla(\d+)". I'm not sure if this line is the right way of processing it, as I think you will get an entire match here rather than just the contents of the capturing group.

Reply:
Should be fixed also